includes/html/api_functions.inc.php

```diff
@@ -2783,6 +2784,36 @@ function get_fdb(Illuminate\Http\Request $request)
     });
 }
 
+function get_nac(Illuminate\Http\Request $request)
+{
+    $hostname = $request->route('hostname');
+
+    if (empty($hostname)) {
+        return api_error(500, 'No hostname has been provided');
+    }
+
+    $device_id = ctype_digit($hostname) ? $hostname : getidbyname($hostname);
+    $device = null;
+    if ($device_id) {
+        // save the current details for returning to the client on successful delete
+        $device = Device::find($device_id);
+    }
+
+    if (! $device) {
```

Comment:
You should copy more modern code here:
```suggestion
    $device = \App\Facades\DeviceCache::get($hostname);

    if (! $device->exists) {
```